### PR TITLE
Feature/cor 261

### PR DESCRIPTION
**Pull request by dev-warner**

### CODE REVIEW

**Review by Chattox — APPROVED** [review]

File: generate-sandpack-config.mjs [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+const examples = ['banner'];
+
+console.log('Sandpack config factory 🏭');
+
+examples.forEach(example => {
+  let __dirname = path.join(example);
+
+  const framworks = fs.readdirSync(__dirname);
+
+  framworks.map(framwork => path.join(__dirname, framwork)).map(createConfig);
```
Comment (srweeks) [inline generate-sandpack-config.mjs]: why map twice?

File: generate-sandpack-config.mjs [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+
+import fs from 'fs';
+import path from 'path';
+
+const examples = ['banner'];
+
+console.log('Sandpack config factory 🏭');
+
+examples.forEach(example => {
+  let __dirname = path.join(example);
```
Comment (srweeks) [inline generate-sandpack-config.mjs]: what do the two underscores mean and can it be a const declaration?

File: generate-sandpack-config.mjs [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+    );
+
+    const parsedPackage = JSON.parse(packageJson);
+    const parsedConfig = JSON.parse(config);
+
+    const { files } = parsedConfig;
+
+    let expandedFiles = [];
+
+    if (_.isArray(files)) {
```
Comment (srweeks) [inline generate-sandpack-config.mjs]: prefer native?

File: generate-sandpack-config.mjs [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+import path from 'path';
+
+const examples = ['banner'];
+
+console.log('Sandpack config factory 🏭');
+
+examples.forEach(example => {
+  let __dirname = path.join(example);
+
+  const framworks = fs.readdirSync(__dirname);
```
Comment (srweeks) [inline generate-sandpack-config.mjs]: typo

File: generate-sandpack-config.mjs [hunk 5]
```diff
@@ -0,0 +1,118 @@
+import _ from 'lodash';
+
+import fs from 'fs';
+import path from 'path';
+
+const examples = ['banner'];
```
Comment (srweeks) [inline generate-sandpack-config.mjs]: Could this be determined without hardcoding? maybe everything in src is considered an example?

File: banner/javascript/sandpack.config.json [hunk 6]
```diff
@@ -0,0 +1,13 @@
+{
+  "files": [
+    "/index.html",
```
Comment (srweeks) [inline banner/javascript/sandpack.config.json]: not sure root relative file paths make sense

File: banner/javascript/sandpack.config.json [hunk 7]
```diff
@@ -0,0 +1,13 @@
...
+    "/src/styles.css",
+    { "title": "/src/index.js", "active": true }
+  ],
+  "customSetup": {
```
Comment (srweeks) [inline banner/javascript/sandpack.config.json]: what is custom about the setup in our custom sandpack setup?

File: banner/javascript/sandpack.config.json [hunk 8]
```diff
@@ -0,0 +1,13 @@
+{
```
Comment (srweeks) [inline banner/javascript/sandpack.config.json]: should we consider a different namespace? Sandpack is a thing and what we're doing is very related but it's not sandpack's doing

**Review by srweeks — COMMENTED** [review]
Possibly we also want to document this new behaviour in the readme?

File: generate-sandpack-config.mjs (hunk 2 above)
Comment (dev-warner) [reply generate-sandpack-config.mjs]: __dirname is normally a predefined node thing but If you are using Node.js modules __dirname and __filename don't exist. so just using that naming scheme

File: generate-sandpack-config.mjs (hunk 5 above)
Comment (dev-warner) [reply generate-sandpack-config.mjs]: i ideally want to move all examples under /lib/ or something similar which would make this easier

File: banner/javascript/sandpack.config.json (hunk 6 above)
Comment (dev-warner) [reply banner/javascript/sandpack.config.json]: these match sandpack configs and are appended using path so thought i'd try match sandpack config

File: banner/javascript/sandpack.config.json (hunk 7 above)
Comment (dev-warner) [reply banner/javascript/sandpack.config.json]: yeah its a sandpack thing

File: banner/javascript/sandpack.config.json (hunk 8 above)
Comment (dev-warner) [reply banner/javascript/sandpack.config.json]: unsure i feel like it describes it best

File: generate-sandpack-config.mjs (hunk 3 above)
Comment (dev-warner) [reply generate-sandpack-config.mjs]: because i'm using isPlainObject might aswell make use of isArray

**Review by pixelscript — APPROVED** [review]

**Review by srweeks — APPROVED** [review]

**Pull request merged**